src/core/tsi/ssl_transport_security.cc

```diff
@@ -1026,28 +1025,128 @@ static int GetCrlFromProvider(X509_STORE_CTX* ctx, X509_CRL** crl_out,
   // and behave how we want for a missing CRL.
   // It is important to treat missing CRLs and empty CRLs differently.
   if (internal_crl == nullptr) {
-    return 0;
+    return absl::NotFoundError("Could not find Crl related to certificate.");
   }
   X509_CRL* crl =
       std::static_pointer_cast<grpc_core::experimental::CrlImpl>(internal_crl)
           ->crl();
 
-  X509_CRL* copy = X509_CRL_dup(crl);
-  *crl_out = copy;
+  return X509_CRL_dup(crl);
+}
+
+// Perform the validation checks in RFC5280 6.3.3 to ensure the given CRL is
+// valid
+static int ValidateCrl(X509* cert, X509* issuer, X509_CRL* crl) {
```

Comment:
Please comment on the return semantics of this function (e.g. return 1 is success, return 0 is failure).

Reply:
Changed to bool and addressed